featuretools/primitives/utils.py

```diff
@@ -126,3 +126,35 @@ def load_primitive_from_file(filepath):
         raise RuntimeError("More than one primitive defined in file %s" % filepath)
 
     return primitives[0]
+
+
+def serialize_primitive(primitive):
+    args_dict = {name: val for name, val in primitive.get_arguments()}
+    cls = type(primitive)
+    return {
+        'type': cls.__name__,
+        'module': cls.__module__,
+        'arguments': args_dict,
+    }
+
+
+def deserialize_primitive(primitive_dict):
+    class_name = primitive_dict['type']
+    module = primitive_dict['module']
+    cls = _find_primitive_class(class_name, module, PrimitiveBase)
+    if not cls:
+        raise RuntimeError('Primitive "%s" in module "%s" not found' %
+                           (class_name, module))
+
+    arguments = primitive_dict['arguments']
+    return cls(**arguments)
+
+
+def _find_primitive_class(class_name, module, current_class):
```

Comment:
I was thinking that the number of primitives should always be relatively small, but with custom primitives it could be a few hundred so the caching could be worth it.